### PR TITLE
Add a missing "genetic algorithm" folder with a basic algorithm inside

### PR DESCRIPTION
**Pull request by D4rkia**
### **Describe your change:**

* [X] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [X] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [X] This pull request is all my own work -- I have not plagiarized.
* [X] I know that pull requests will not be merged if they fail the automated tests.
* [X] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [ ] All new Python files are placed inside an existing directory.
* [X] All filenames are in all lowercase characters with no spaces or dashes.
* [X] All functions and variable names follow Python naming conventions.
* [X] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [X] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [X] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [X] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: genetic_algorithm/basic_string.py [hunk 1]
```diff
@@ -0,0 +1,148 @@
+"""
+Simple multithreaded algorithm to show how the 4 phases of a genetic
+algorithm works (Evaluation, Selection, Crossover and Mutation)
+
+https://en.wikipedia.org/wiki/Genetic_algorithm
+
+
```
Comment (cclauss) [inline genetic_algorithm/basic_string.py:7]:
```suggestion
```

File: genetic_algorithm/basic_string.py [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+Author: D4rkia
+"""
+
+from typing import List, Tuple
+import random
```
Comment (cclauss) [inline genetic_algorithm/basic_string.py:11]:
```suggestion
import random
from typing import List, Tuple
```
We use `isort` to sort imports.

File: genetic_algorithm/basic_string.py [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+
+N_POPULATION = 200
+N_SELECTED = 50
+MUTATION_PROBABILITY = 0.1
+random.seed(random.randint(0, 1000))
```
Comment (cclauss) [inline genetic_algorithm/basic_string.py:16]: Please add a comment on this line or just above to explain why this command is useful.

File: genetic_algorithm/basic_string.py [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+
+
+def basic(sentence: str, genes: List[str]) -> Tuple[int, int, str]:
+    """The actual algorithm"""
+    # Verify that the sentence contains no other genes
+    # that the ones inside genes variable
```
Comment (cclauss) [inline genetic_algorithm/basic_string.py:22]:
```suggestion
    """
    Verify that the sentence contains no genes besides the ones inside genes variable.
    """
```

File: genetic_algorithm/basic_string.py [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+    for i in sentence:
+        if i not in genes_list:
+            raise ValueError(f"{i} is not in  genes list, evolution can't converge")
```
Comment (cclauss) [inline genetic_algorithm/basic_string.py:25]:
```suggestion
    if any(c not in genes_list for c in sentence):
        raise ValueError(f"{c} is not in genes list, evolution can't converge")
```
More readable to use `c` for characters and `i` for integers.

File: genetic_algorithm/basic_string.py [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+        total_population += len(population)
+
+        # Random population created now it's time to evaluate
+        def evaluate(item: str, main_sentence: str = sentence) -> Tuple[str, float]:
+            """
+                Evaluate how similar the item is with the sentence by just
+                counting each char in the right position
+
+                >>> evaluate("Helxo Worlx", Hello World)
+                ["Helxo Worlx", 9]
```
Comment (cclauss) [inline genetic_algorithm/basic_string.py:49]:
```suggestion
            Evaluate how similar the item is with the sentence by just
            counting each char in the right position

            >>> evaluate("Helxo Worlx", Hello World)
            ["Helxo Worlx", 9]
```

File: genetic_algorithm/basic_string.py [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+            """
+            score = 0
+            for position, g in enumerate(item):
+                if g == main_sentence[position]:
+                    score += 1
```
Comment (cclauss) [inline genetic_algorithm/basic_string.py:54]:
```suggestion
            score = len([g for position, g in enumerate(item) if g == main_sentence[position]])
```

File: genetic_algorithm/basic_string.py [hunk 8]
```diff
@@ -0,0 +1,147 @@
...
+    genes_list = list(
+        " ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklm"
+        "nopqrstuvwxyz.,;!?+-*#@^'èéòà€ù=)(&%$£/\\"
+    )
+    print(
+        "\nGeneration: %s\nTotal Population: %s\nSentence: %s"
+        % basic(sentence_str, genes_list)
+    )
+
+__author__ = "D4rkia.Fudo"
```
Comment (cclauss) [inline genetic_algorithm/basic_string.py:147]: What is the utility of this line?

File: genetic_algorithm/basic_string.py (hunk 5 above)
Comment (D4rkia) [reply genetic_algorithm/basic_string.py]: This gives me error with flake8 so it fail automated tests.

`F821 undefined name 'c'`

**Comment by D4rkia** [comment]
> Usually genetic algorithms are evolving `code` (is that a better variable name than `sentence`?) towards some goal or objective. What state is each generation evolving towards?

You're right, I replaced it with `target`

File: genetic_algorithm/basic_string.py [hunk 9]
```diff
@@ -0,0 +1,164 @@
...
+            # if this check is disabled
+            # the algorithm will take forever to compute large strings
+            # but will also calcolate small string in a lot less generations
+            if len(population) > N_POPULATION:
+                break
+
+
+if __name__ == "__main__":
+    target_str = (
+        "This is a genetic algorithm to evaluate, combine, evolve  mutate a string!"
```
Comment (cclauss) [inline genetic_algorithm/basic_string.py:155]:
```suggestion
        "This is a genetic algorithm to evaluate, combine, evolve, and mutate a string!"
```

File: genetic_algorithm/basic_string.py [hunk 10]
```diff
@@ -0,0 +1,164 @@
...
+            return "".join(child_list)
+
+        # This is Selection
+        for i in range(N_SELECTED):
+            population.extend(select(population_score[int(i)]))
+            # Check if the population has already reached the maximum value and
+            # break the cycle
+            # if this check is disabled
+            # the algorithm will take forever to compute large strings
+            # but will also calcolate small string in a lot less generations
```
Comment (cclauss) [inline genetic_algorithm/basic_string.py:148]:
```suggestion
            # but will also calculate small string in a lot fewer generations
```

File: genetic_algorithm/basic_string.py (hunk 3 above)
Comment (D4rkia) [reply genetic_algorithm/basic_string.py:25]: Fixed

**Comment by cclauss** [comment]
We kinda forgot to add [`string.digits`](https://docs.python.org/3/library/string.html) to `genes` but we can leave that as an exercise for the reader ;-)

**Pull request merged**